### PR TITLE
Update rubocop to latest and fix as many violations as possible

### PR DESCRIPTION
**Pull request by lstrzebinczyk**
As in the title. We were able to fix about 1000 violations in automated manner, I manually fixed a bunch others, but hundreds of violations remain. I created a todo file to cut the work at where it's right now.

granite-form codebase is in a poor shape.

**AC:**
  - Trust specs

### CODE REVIEW

**Comment by lstrzebinczyk** [comment]
@toptal-anvil ping reviewers

**Review by kml — APPROVED** [review]

**Review by sasha370 — APPROVED** [review]
LGTM :+1:

File: spec/granite/form/types/date_time_spec.rb [hunk 1]
```diff
@@ -3,7 +3,6 @@
 require 'spec_helper'
 
 RSpec.describe Granite::Form::Types::DateTime do
-  # rubocop:disable Style/DateTime
```
Comment (mbie) [inline spec/granite/form/types/date_time_spec.rb:6]: There is `# rubocop:enable Style/DateTime` in line 19 that could be removed.

File: spec/granite/form/types/time_spec.rb [hunk 2]
```diff
@@ -3,7 +3,6 @@
 require 'spec_helper'
 
 RSpec.describe Granite::Form::Types::Time do
-  # rubocop:disable Style/DateTime
```
Comment (mbie) [inline spec/granite/form/types/time_spec.rb:6]: ditto

**Review by mbie — APPROVED** [review]
Please check the suggestion

File: lib/granite/form/model/associations/nested_attributes.rb [hunk 3]
```diff
@@ -93,38 +112,44 @@ def self.assign_nested_attributes_for_one_to_one_association(object, association
...
-                attributes_collection = if keys.include?(primary_attribute_name) || keys.include?(primary_attribute_name.to_sym)
-                  [attributes_collection]
-                else
-                  attributes_collection.values
-                end
+                attributes_collection = if keys.include?(primary_attribute_name) || keys.include?(primary_attribute_name.to_sym) # rubocop:disable Layout/LineLength
+                                          [attributes_collection]
+                                        else
+                                          attributes_collection.values
+                                        end
```
Comment (id-ilych) [inline lib/granite/form/model/associations/nested_attributes.rb]:
```suggestion
                attributes_collection =
                  if keys.include?(primary_attribute_name) || keys.include?(primary_attribute_name.to_sym)
                    [attributes_collection]
                  else
                    attributes_collection.values
                  end
```

File: spec/granite/form/active_record/associations_spec.rb [hunk 4]
```diff
@@ -194,13 +207,24 @@ def read_attribute(name)
 
     specify { expect(User.reflect_on_association(:projects).klass).to eq(User::Project) }
     specify { expect(User.new.projects).to eq([]) }
-    specify { expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded) }
-    specify { expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects).to match([have_attributes(title: 'Project')]) }
+
+    specify do
+      expect(User.new.tap do |u|
+               u.projects.build(title: 'Project')
+             end.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded)
```
Comment (id-ilych) [inline spec/granite/form/active_record/associations_spec.rb]:
```suggestion
      expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects)
        .to be_a(Granite::Form::Model::Associations::Collection::Embedded)
```

File: spec/granite/form/active_record/associations_spec.rb [hunk 5]
```diff
@@ -194,13 +207,24 @@ def read_attribute(name)
...
+    end
+
+    specify do
+      expect(User.new.tap do |u|
+               u.projects.build(title: 'Project')
+             end.projects).to match([have_attributes(title: 'Project')])
```
Comment (id-ilych) [inline spec/granite/form/active_record/associations_spec.rb]:
```suggestion
      expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects)
        .to match([have_attributes(title: 'Project')])
```

File: spec/granite/form/model/associations/embeds_many_spec.rb [hunk 6]
```diff
@@ -210,27 +216,41 @@
...
     let(:existing_user) { User.instantiate name: 'Rick' }
 
     specify { expect(association.target.map(&:title)).to eq(['Default']) }
     specify { expect(association.target.map(&:new_record?)).to eq([true]) }
-    specify { expect { association.replace([new_project]) }.to change { association.target.map(&:title) }.to eq(['Project']) }
+
+    specify do
+      expect { association.replace([new_project]) }.to change {
+                                                         association.target.map(&:title)
+                                                       }.to eq(['Project'])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_many_spec.rb]:
```suggestion
      expect { association.replace([new_project]) }
        .to change { association.target.map(&:title) }
        .to eq(['Project'])
```

File: spec/granite/form/model/associations/embeds_many_spec.rb [hunk 7]
```diff
@@ -210,27 +216,41 @@
...
     specify { expect { association.replace([]) }.to change { association.target }.to([]) }
 
     specify { expect(existing_association.target).to eq([]) }
-    specify { expect { existing_association.replace([new_project]) }.to change { existing_association.target.map(&:title) }.to(['Project']) }
-    specify { expect { existing_association.replace([]) }.not_to change { existing_association.target } }
+
+    specify do
+      expect { existing_association.replace([new_project]) }.to change {
+                                                                  existing_association.target.map(&:title)
+                                                                }.to(['Project'])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_many_spec.rb]:
```suggestion
      expect { existing_association.replace([new_project]) }
        .to change { existing_association.target.map(&:title) }
        .to(['Project'])
```

File: spec/granite/form/model/associations/embeds_many_spec.rb [hunk 8]
```diff
@@ -342,37 +367,46 @@
...
+
     specify do
-      expect { muffle(Granite::Form::AssociationTypeMismatch) { existing_association.writer([new_project1, Dummy.new, new_project2]) } }
-        .not_to change { existing_association.reader }
+      expect do
+        muffle(Granite::Form::AssociationTypeMismatch) do
+          existing_association.writer([new_project1, Dummy.new, new_project2])
+        end
+      end
+        .not_to(change { existing_association.reader })
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_many_spec.rb]: It's hard to read, could it be combined with the previous one as `.to raise_error ... .and(not_to(change ...))`?

File: spec/granite/form/model/associations/embeds_many_spec.rb [hunk 9]
```diff
@@ -392,39 +426,48 @@
...
     specify do
-      expect { muffle(Granite::Form::AssociationTypeMismatch) { existing_association.concat(new_project1, Dummy.new, new_project2) } }
+      expect do
+        muffle(Granite::Form::AssociationTypeMismatch) do
+          existing_association.concat(new_project1, Dummy.new, new_project2)
+        end
+      end
         .to change { existing_association.reader.map(&:attributes) }
-        .from([{'title' => 'Genesis'}]).to([{'title' => 'Genesis'}, {'title' => 'Project 1'}])
+        .from([{ 'title' => 'Genesis' }]).to([{ 'title' => 'Genesis' }, { 'title' => 'Project 1' }])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_many_spec.rb:462]: It's hard to read, could it be combined with the previous one as `.to raise_error ... .and(not_to(change ...))`?

File: spec/granite/form/model/associations/embeds_many_spec.rb [hunk 10]
```diff
@@ -392,39 +426,48 @@
...
       expect(existing_association.concat(new_project1, new_project2))
         .to eq([existing_user.projects.first, new_project1, new_project2])
     end
+
     specify do
       expect { existing_association.concat([new_project1, new_project2]) }
         .to change { existing_association.reader.map(&:attributes) }
-        .from([{'title' => 'Genesis'}]).to([{'title' => 'Genesis'}, {'title' => 'Project 1'}, {'title' => 'Project 2'}])
+        .from([{ 'title' => 'Genesis' }]).to([{ 'title' => 'Genesis' }, { 'title' => 'Project 1' },
+                                              { 'title' => 'Project 2' }])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_many_spec.rb]:
```suggestion
        .from([{ 'title' => 'Genesis' }]).to([{ 'title' => 'Genesis' },
                                              { 'title' => 'Project 1' },
                                              { 'title' => 'Project 2' }])
```

File: spec/granite/form/model/associations/embeds_one_spec.rb [hunk 11]
```diff
@@ -191,11 +195,18 @@ def after_add(object)
     specify { expect { association.replace(nil) }.to change { association.target }.to be_nil }
 
     specify { expect(existing_association.target).to be_nil }
-    specify { expect { existing_association.replace(new_author) }.to change { existing_association.target }.to(an_instance_of(Author)) }
-    specify { expect { existing_association.replace(nil) }.not_to change { existing_association.target } }
+
+    specify do
+      expect { existing_association.replace(new_author) }.to change {
+                                                               existing_association.target
+                                                             }.to(an_instance_of(Author))
```
Comment (id-ilych) [inline spec/granite/form/model/associations/embeds_one_spec.rb]:
```suggestion
      expect { existing_association.replace(new_author) }
        .to change { existing_association.target }
        .to(an_instance_of(Author))
```

File: spec/granite/form/model/associations/references_many_spec.rb [hunk 12]
```diff
@@ -142,41 +152,57 @@
...
+
     specify do
-      expect { muffle(Granite::Form::AssociationTypeMismatch) { existing_association.writer([new_author1, Dummy.new, new_author2]) } }
-        .not_to change { existing_association.reader }
+      expect do
+        muffle(Granite::Form::AssociationTypeMismatch) do
+          existing_association.writer([new_author1, Dummy.new, new_author2])
+        end
+      end
+        .not_to(change { existing_association.reader })
```
Comment (id-ilych) [inline spec/granite/form/model/associations/references_many_spec.rb:171]: Same here

File: spec/granite/form/model/associations/references_many_spec.rb [hunk 13]
```diff
@@ -222,13 +251,23 @@
...
         .from([author.id]).to([author.id, new_author1.id])
     end
+
     specify do
-      expect { muffle(Granite::Form::AssociationTypeMismatch) { existing_association.concat(new_author1, Dummy.new, new_author2) } }
+      expect do
+        muffle(Granite::Form::AssociationTypeMismatch) do
+          existing_association.concat(new_author1, Dummy.new, new_author2)
+        end
+      end
```
Comment (id-ilych) [inline spec/granite/form/model/associations/references_many_spec.rb:270]: Same here

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb [hunk 14]
```diff
@@ -55,43 +56,60 @@
...
       context do
         let(:other) { Project.new title: 'Other' }
+
         before { user.projects = [other] }
-        specify { expect { user.projects.concat project }.to change { user.projects }.from([other]).to([other, project]) }
+
+        specify do
+          expect { user.projects.concat project }.to change {
+                                                       user.projects
+                                                     }.from([other]).to([other, project])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/embeds_many_spec.rb]:
```suggestion
          expect { user.projects.concat project }
            .to change { user.projects }
            .from([other])
            .to([other, project])
```

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb [hunk 15]
```diff
@@ -112,8 +130,18 @@
 
       specify { expect(User.reflect_on_association(:projects).klass).to eq(User::Project) }
       specify { expect(User.new.projects).to eq([]) }
-      specify { expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded) }
-      specify { expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects).to match([have_attributes(title: 'Project')]) }
+
+      specify do
+        expect(User.new.tap do |u|
+                 u.projects.build(title: 'Project')
+               end.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded)
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/embeds_many_spec.rb]:
```suggestion
        expect(User.new.tap { |u| u.projects.build(title: 'Project') }.projects)
          .to be_a(Granite::Form::Model::Associations::Collection::Embedded)
```
or
```suggestion
        user = User.new
        user.projects.build(title: 'Project')
        expect(user.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded)
```

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb [hunk 16]
```diff
@@ -112,8 +130,18 @@
...
+      end
+
+      specify do
+        expect(User.new.tap do |u|
+                 u.projects.build(title: 'Project')
+               end.projects).to match([have_attributes(title: 'Project')])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/embeds_many_spec.rb]: Same here

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb [hunk 17]
```diff
@@ -130,8 +158,18 @@
...
+      specify do
+        expect(User.new.tap do |u|
+                 u.projects.build(title: 'Project')
+               end.projects).to be_a(Granite::Form::Model::Associations::Collection::Embedded)
+      end
+
+      specify do
+        expect(User.new.tap do |u|
+                 u.projects.build(title: 'Project')
+               end.projects).to match([have_attributes(title: 'Project', value: nil)])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/embeds_many_spec.rb]: Same here

File: spec/granite/form/model/associations/reflections/embeds_one_spec.rb [hunk 18]
```diff
@@ -104,7 +109,12 @@
       specify { expect(Book.reflect_on_association(:author).klass).to eq(Book::Author) }
       specify { expect(Book.new.author).to be_nil }
       specify { expect(Book.new.tap { |b| b.build_author(name: 'Author') }.author).to be_a(Book::Author) }
-      specify { expect(Book.new.tap { |b| b.build_author(name: 'Author') }.author).to have_attributes(name: 'Author', age: nil) }
+
+      specify do
+        expect(Book.new.tap do |b|
+                 b.build_author(name: 'Author')
+               end.author).to have_attributes(name: 'Author', age: nil)
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/embeds_one_spec.rb]: Same here

File: spec/granite/form/model/associations/reflections/references_many_spec.rb [hunk 19]
```diff
@@ -139,7 +142,13 @@
       specify { expect { book.owners = [other] }.to change { book.owner_ids }.from([nil]).to([other.id]) }
       specify { expect { book.owners = [other] }.to change { book.owners }.from([an_instance_of(Author)]).to([other]) }
       specify { expect { book.owner_ids = [other.id] }.to change { book.owner_ids }.from([nil]).to([other.id]) }
-      specify { expect { book.owner_ids = [other.id] }.to change { book.owners }.from([an_instance_of(Author)]).to([other]) }
+
+      specify do
+        expect { book.owner_ids = [other.id] }.to change {
+                                                    book.owners
+                                                  }.from([an_instance_of(Author)]).to([other])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/references_many_spec.rb]:
```suggestion
        expect { book.owner_ids = [other.id] }
          .to change { book.owners }
          .from([an_instance_of(Author)])
          .to([other])
```

File: spec/granite/form/model/associations/reflections/references_many_spec.rb [hunk 20]
```diff
@@ -249,8 +266,18 @@
...
+    xit do
+      expect { book_with_author.author_ids << other.id }.to change {
+                                                              book_with_author.authors
+                                                            }.from([author]).to([author, other])
+    end
+
+    it {
+      expect { book_with_author.author_ids = [other.id] }.to change {
+                                                               book_with_author.authors
+                                                             }.from([author]).to([other])
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/references_many_spec.rb]: Same here

File: spec/granite/form/model/associations/reflections/references_many_spec.rb [hunk 21]
```diff
@@ -289,12 +320,27 @@
...
 
     context do
       before { book.authors = [other] }
-      specify { expect { book.author_ids = [author.id] }.to change { book.author_ids }.from([other.id]).to([author.id]) }
+
+      specify do
+        expect { book.author_ids = [author.id] }.to change {
+                                                      book.author_ids
+                                                    }.from([other.id]).to([author.id])
+      end
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/references_many_spec.rb]: Same here

File: spec/granite/form/model/associations/reflections/references_one_spec.rb [hunk 22]
```diff
@@ -234,29 +246,39 @@
...
+
     specify { expect { book.author_id = author.id }.to change { book.author_id }.from(nil).to(author.id) }
     specify { expect { book.author_id = author.id }.to change { book.author }.from(nil).to(author) }
     specify { expect { book.author_id = author }.to change { book.author }.from(nil).to(author) }
 
-    specify { expect { book.author_id = author.id.next.to_s }.to change { book.author_id }.from(nil).to(author.id.next) }
+    specify do
+      expect { book.author_id = author.id.next.to_s }.to change {
+                                                           book.author_id
+                                                         }.from(nil).to(author.id.next)
```
Comment (id-ilych) [inline spec/granite/form/model/associations/reflections/references_one_spec.rb]: Same here

File: spec/granite/form/model/attributes/attribute_spec.rb [hunk 23]
```diff
@@ -5,12 +5,17 @@
...
+                        { type: Object }.merge(options))
     Dummy.new.attribute(:field)
   end
 
   describe '#read' do
-    let(:field) { attribute(type: String, normalizer: ->(v) { v ? v.strip : v }, default: :world, enum: %w[hello 42 world]) }
+    let(:field) do
+      attribute(type: String, normalizer: lambda { |v|
+                                            v ? v.strip : v
+                                          }, default: :world, enum: %w[hello 42 world])
```
Comment (id-ilych) [inline spec/granite/form/model/attributes/attribute_spec.rb]:
```suggestion
      normalizer = ->(v) { v ? v.strip : v }
      attribute(type: String, normalizer: normalizer, default: :world, enum: %w[hello 42 world])
```

File: spec/granite/form/model/attributes/attribute_spec.rb [hunk 24]
```diff
@@ -36,7 +41,11 @@ def attribute(*args)
     specify { expect(field.tap { |r| r.write('') }.read_before_type_cast).to eq('') }
 
     context ':readonly' do
-      specify { expect(attribute(readonly: true, default: :world).tap { |r| r.write('string') }.read_before_type_cast).to eq(:world) }
+      specify do
+        expect(attribute(readonly: true, default: :world).tap do |r|
+                 r.write('string')
+               end.read_before_type_cast).to eq(:world)
```
Comment (id-ilych) [inline spec/granite/form/model/attributes/attribute_spec.rb]:
```suggestion
        attr = attribute(readonly: true, default: :world)
        attr.write('string')
        expect(attr.read_before_type_cast).to eq(:world)
```

File: spec/granite/form/model/attributes/attribute_spec.rb [hunk 25]
```diff
@@ -58,11 +67,22 @@ def attribute(*args)
...
+    specify do
+      expect(attribute(normalizer: [->(v) { v.strip }, lambda { |v|
+                                                         v.first(4)
+                                                       }]).normalize(' hello ')).to eq('hell')
+    end
+
+    specify do
+      expect(attribute(normalizer: [->(v) { v.first(4) }, lambda { |v|
+                                                            v.strip
+                                                          }]).normalize(' hello ')).to eq('hel')
```
Comment (id-ilych) [inline spec/granite/form/model/attributes/attribute_spec.rb]:
```suggestion
    specify do
      normalizers = [->(v) { v.strip }, ->(v) { v.first(4) }]
      attr = attribute(normalizer: normalizers)
      expect(attr.normalize(' hello ')).to eq('hell')
    end

    specify do
      normalizers = [->(v) { v.first(4) }, ->(v) { v.strip }]
      attr = attribute(normalizer: normalizers)
      expect(attr.normalize(' hello ')).to eq('hel')
```

File: spec/granite/form/model/attributes/base_spec.rb [hunk 26]
```diff
@@ -7,12 +7,17 @@
...
+                        options.reverse_merge(type: Object))
     model.attribute(:field)
   end
 
   describe '#read' do
-    let(:field) { attribute(type: String, normalizer: ->(v) { v ? v.strip : v }, default: :world, enum: %w[hello 42 world]) }
+    let(:field) do
+      attribute(type: String, normalizer: lambda { |v|
+                                            v ? v.strip : v
+                                          }, default: :world, enum: %w[hello 42 world])
```
Comment (id-ilych) [inline spec/granite/form/model/attributes/base_spec.rb]:
```suggestion
      normalizer = ->(v) { v ? v.strip : v }
      attribute(type: String, normalizer: normalizer, default: :world, enum: %w[hello 42 world])
```

File: spec/granite/form/model/attributes_spec.rb [hunk 27]
```diff
@@ -349,21 +373,49 @@ def with_assigned_value(value)
...
+                                                        })
+      end
+
+      specify do
+        with_assigned_value(from: 3,
+                            to: 1).to have_attributes(dict: { 'from' => 1, 'to' => 3 },
+                                                      dict_before_type_cast: {
+                                                        from: 3, to: 1
+                                                      })
+      end
```
Comment (id-ilych) [inline spec/granite/form/model/attributes_spec.rb]: Please consider more readable formatting

File: spec/granite/form/model/primary_spec.rb [hunk 28]
```diff
@@ -63,20 +65,33 @@
...
+      expect(model.new(name: 'Hello').tap { |o| o.id = 1 }).not_to eq(model.new(name: 'Hello').tap do |o|
+                                                                        o.id = 2
+                                                                      end)
+    end
+
+    specify do
+      expect(model.new(name: 'Hello').tap { |o| o.id = 1 }).not_to eql(model.new(name: 'Hello').tap do |o|
+                                                                         o.id = 2
+                                                                       end)
+    end
```
Comment (id-ilych) [inline spec/granite/form/model/primary_spec.rb:79]: Same here

File: spec/granite/form/model/scopes_spec.rb [hunk 29]
```diff
@@ -45,7 +45,12 @@ def hidden_method() end
       let(:scope) { model.scope([model.new(name: 'Hello'), model.new(name: 'World'), model.new(name: 'Mars')]) }
 
       specify { expect(scope).to be_instance_of model.scope_class }
-      specify { expect { model.scope([model.new(name: 'Hello'), {}]) }.to raise_error Granite::Form::AssociationTypeMismatch }
+
+      specify do
+        expect do
+          model.scope([model.new(name: 'Hello'), {}])
+        end.to raise_error Granite::Form::AssociationTypeMismatch
```
Comment (id-ilych) [inline spec/granite/form/model/scopes_spec.rb:52]:
```suggestion
        expect { model.scope([model.new(name: 'Hello'), {}]) }
          .to raise_error Granite::Form::AssociationTypeMismatch
```

File: spec/support/shared/nested_attribute_examples.rb [hunk 30]
```diff
@@ -23,54 +23,113 @@
     let(:user) { User.new }
 
     specify { expect { user.profile_attributes = {} }.to change { user.profile }.to(an_instance_of(Profile)) }
-    specify { expect { user.profile_attributes = {first_name: 'User'} }.to change { user.profile.try(:first_name) }.to('User') }
-    specify { expect { user.profile_attributes = {identifier: 42, first_name: 'User'} }.to raise_error Granite::Form::ObjectNotFound }
+
+    specify do
+      expect { user.profile_attributes = { first_name: 'User' } }.to change {
+                                                                       user.profile.try(:first_name)
+                                                                     }.to('User')
```
Comment (id-ilych) [inline spec/support/shared/nested_attribute_examples.rb]:
```suggestion
      expect { user.profile_attributes = { first_name: 'User' } }
        .to change { user.profile.try(:first_name) }
        .to('User')
```

File: spec/support/shared/nested_attribute_examples.rb [hunk 31]
```diff
@@ -23,54 +23,113 @@
...
+          user.profile_attributes = { first_name: 'User 1', _destroy: '1' }
+        end.not_to(change { user.profile.first_name })
+      end
+
+      specify do
+        expect do
+          user.profile_attributes = { identifier: profile.identifier.to_s, first_name: 'User 1', _destroy: '1' }
+        end.to change {
+                 user.profile.first_name
+               }.to('User 1')
```
Comment (id-ilych) [inline spec/support/shared/nested_attribute_examples.rb]: Please improve readability

File: spec/support/shared/nested_attribute_examples.rb [hunk 32]
```diff
@@ -23,54 +23,113 @@
...
+        end
+
         specify do
           expect do
-            user.profile_attributes = {first_name: 'User 1', _destroy: '1'}
-          end.not_to change { user.profile.first_name }
+            user.profile_attributes = { identifier: profile.identifier.to_s, first_name: 'User 1', _destroy: '1' }
+          end.to change {
+                   user.profile
+                 }.to(nil)
```
Comment (id-ilych) [inline spec/support/shared/nested_attribute_examples.rb]: Please improved readability

File: spec/support/shared/nested_attribute_examples.rb [hunk 33]
```diff
@@ -94,14 +153,19 @@
       end
 
       specify { expect { user.profile_attributes = {} }.to change { user.profile }.to(an_instance_of(Profile)) }
-      specify { expect { user.profile_attributes = {first_name: 'User'} }.to change { user.profile.try(:first_name) }.to('User') }
+
+      specify do
+        expect { user.profile_attributes = { first_name: 'User' } }.to change {
+                                                                         user.profile.try(:first_name)
+                                                                       }.to('User')
```
Comment (id-ilych) [inline spec/support/shared/nested_attribute_examples.rb]: Same here

File: lib/granite/form/model/associations/base.rb [hunk 34]
```diff
@@ -66,10 +68,11 @@ def transaction
           end
 
           def inspect
-            "#<#{reflection.macro.to_s.camelize} #{target.inspect.truncate(50, omission: collection? ? '...]' : '...')}>"
+            "#<#{reflection.macro.to_s.camelize} #{target.inspect.truncate(50,
+                                                                           omission: collection? ? '...]' : '...')}>"
```
Comment (id-ilych) [inline lib/granite/form/model/associations/base.rb]:
```suggestion
            omission = collection? ? '...]' : '...')
            "#<#{reflection.macro.to_s.camelize} #{target.inspect.truncate(50, omission: omission}>"
```

**Review by id-ilych — COMMENTED** [review]
Autofix did a mostly good job, but some human touch could be nice

File: lib/granite/form/model/associations/nested_attributes.rb (hunk 3 above)
Comment (kml) [reply lib/granite/form/model/associations/nested_attributes.rb]: Not sure on the suggestion. As first we apply rule that introduces indentation, then try to workaround it by moving `if`.
Why not disable the rule or apply to it without workarounds?

File: spec/granite/form/model/associations/embeds_many_spec.rb (hunk 8 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/embeds_many_spec.rb]: The previous one doesn't do `muffle`. I've removed the one line here, I don't think we can improve much more without diving deep into the spec

File: spec/granite/form/model/associations/embeds_many_spec.rb (hunk 9 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/embeds_many_spec.rb:462]: Improved :+1:

File: spec/granite/form/model/associations/references_many_spec.rb (hunk 12 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/references_many_spec.rb:171]: I would argue that changing those `muffle` cases isn't making anything more readable. I would leave those cases as they are.

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb (hunk 16 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/embeds_many_spec.rb]: :+1:

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb (hunk 17 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/embeds_many_spec.rb]: :+1:

File: spec/granite/form/model/associations/references_many_spec.rb (hunk 13 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/references_many_spec.rb:270]: :+1:

File: spec/granite/form/model/associations/reflections/embeds_many_spec.rb (hunk 16 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/embeds_many_spec.rb]: :+1:

File: spec/granite/form/model/associations/reflections/embeds_one_spec.rb (hunk 18 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/embeds_one_spec.rb]: :+1:

File: spec/granite/form/model/associations/reflections/references_many_spec.rb (hunk 20 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/references_many_spec.rb]: :+1:

File: spec/granite/form/model/associations/reflections/references_many_spec.rb (hunk 21 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/references_many_spec.rb]: :+1:

File: spec/granite/form/model/associations/reflections/references_one_spec.rb (hunk 22 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/associations/reflections/references_one_spec.rb]: :+1:

File: spec/granite/form/model/attributes_spec.rb (hunk 27 above)
Comment (lstrzebinczyk) [reply spec/granite/form/model/attributes_spec.rb]: :+1:

**Review by id-ilych — APPROVED** [review]

**Pull request merged**